spec/rubocop/cop/layout/heredoc_indentation_spec.rb

```diff
@@ -111,142 +21,170 @@ def foo
+          RUBY2
+          end
+        RUBY
+      end
+
+      it 'accepts for an empty line' do
+        expect_no_offenses(<<~RUBY)
+          <<-#{quote}RUBY2#{quote}
+
+          RUBY2
+        RUBY
+      end
 
       context 'when Layout/LineLength is configured' do
         let(:allow_heredoc) { false }
 
-        include_examples 'offense', 'short heredoc', <<~RUBY, <<~CORRECTION
-          <<#{quote}RUBY2#{quote}
-          12
+        it 'accepts for long heredoc' do
+          expect_no_offenses(<<~RUBY)
+            <<#{quote}RUBY2#{quote}
+            12345678
+            RUBY2
+          RUBY
+        end
+      end
+
+      it 'registers an offense for not indented' do
+        expect_offense(<<~RUBY)
+          <<~#{quote}RUBY2#{quote}
+          something
+          ^^^^^^^^^ Use 2 spaces for indentation in a heredoc.
           RUBY2
         RUBY
-          <<#{quote}RUBY2#{quote}.strip_indent
-            12
+
+        expect_correction(<<~CORRECTION)
+          <<~#{quote}RUBY2#{quote}
+            something
           RUBY2
         CORRECTION
+      end
 
-        include_examples 'accept', 'long heredoc', <<~RUBY
-          <<#{quote}RUBY2#{quote}
-          12345678
-          RUBY2
+      it 'registers an offense for minus level indented' do
+        expect_offense(<<~RUBY)
+          def foo
+            <<~#{quote}RUBY2#{quote}
+          something
+          ^^^^^^^^^ Use 2 spaces for indentation in a heredoc.
+            RUBY2
+          end
         RUBY
+
+        expect_correction(<<~CORRECTION)
+          def foo
+            <<~#{quote}RUBY2#{quote}
+              something
+            RUBY2
+          end
+        CORRECTION
       end
 
-      it 'displays a message with suggestion powerpack' do
+      it 'registers an offense for too deep indented' do
         expect_offense(<<~RUBY)
-          <<-RUBY2
-          foo
-          ^^^ Use 2 spaces for indentation in a heredoc by using `String#strip_indent`.
+          <<~#{quote}RUBY2#{quote}
+              something
+          ^^^^^^^^^^^^^ Use 2 spaces for indentation in a heredoc.
           RUBY2
         RUBY
-      end
-
-      context 'EnforcedStyle is `squiggly`' do
-        let(:cop_config) do
-          { 'EnforcedStyle' => :squiggly }
-        end
 
-        include_examples 'offense', 'not indented', <<~RUBY, <<~CORRECTION
+        expect_correction(<<~CORRECTION)
           <<~#{quote}RUBY2#{quote}
-          something
+            something
+          RUBY2
+        CORRECTION
+      end
+
+      it 'registers an offense for not indented, without `~`' do
+        annotated_source = <<~RUBY
+          <<#{quote}RUBY2#{quote}
+          foo
+          ^^^ Use 2 spaces for indentation in a heredoc by using `<<~` instead of `<<`.
           RUBY2
         RUBY
+
+        expect_offense(annotated_source)
+
+        corrected = looped_autocorrect_from_annotated_source(annotated_source)
+        expect(corrected).to eq(<<~CORRECTION)
```

Comment:
It would actually be pretty neat if you could pass a `loop: true` 2nd argument to `expect_correction`. I noticed a few other places in the specs that rely on multiple passes when testing autocorrection.

Reply:
You can open a separate ticket to track this idea.